aip/general/0193.md

```diff
@@ -62,11 +62,90 @@ a [`PreconditionFailure`][PreconditionFailure].
 **Note:** `ErrorInfo` represents a special case. There **must** be exactly one
 `ErrorInfo`. It is required.
 
-### Localization
+### Error messages
 
-Error messages **must** be in English. If a localized error message is also
-required, the service **should** use [`google.rpc.LocalizedMessage`][details]
-in the `details` field.
+The [`google.rpc.Status`][Status] error payload **should** contain two
+error messages.
+
+1. The payload **must** contain an error message in the `message` field
+   found at the root of [`Status`][Status].
+1. The payload **should** contain
+   [`google.rpc.LocalizedMessage`][LocalizedMessage] in `details` on
+   `Status`. This is the location of the second error message.
+
+The error message on `Status` should,
+
+- Be developer-facing, human-readable, and in English
+- Both expalin the error and offer an actionable resolution to it
+  ([citation](https://cloud.google.com/apis/design/errors#error_model))
+
+Once launched the value of `message` at the root-level of `Status`
+**must not** change for the error payload to be backwards compatible.
+
+When including `LocalizedMessage`, both the `locale` and `message`
+fields **must** be present.
+
+#### Changing error messages
+
+A new, additional error message **may** be added because the error
+message at the root of the payload **must not** change once launched.
+This second error messages allows for continuous, iterative
+improvement. It is potentially displayed directly to end users by
+client application.
+
+When adding a new error messsage with `LocalizedMessage`, `ErrorInfo`
+**must** be introduced before or concurrently. If there are dyanmic
+variables found in the message, these variables must be included in the
```

Comment:
In which message? The LocalizedMessage.message i.e. the "new" message? or the "old" one?

Reply:
I clarified this by saying that `ErrorInfo` is required for new errors and that variables **must** be in `metadata`. Feel free to reopen.